Fixes to the ClusterServiceVersion; now working as intended.

cunei opened the pull request:
A number of changes are included in this pull request:
- The 'required' section of the owned crds was redundant; also,
  parsing it would cause a bizarre "couldn't parse plural.group"
  error message
- The description and displayName are required for all the elements
  of specDescriptors and statusDescriptors (they were in the wrong
  place). Also, tried to give informative description strings
- the 'clusterPermissions' section was set to null; it needs to
  be a (possibly empty) array, instead
- the serviceAccount permissions were set in the 'permissions'
  section, which translates to RBAC roleBindings. This is not
  sufficient, since the operator needs to be able to work on
  namespaces other than the one it is deployed to. So, moved
  those definitions to 'clusterPermissions', instead, which
  get translated to clusterRoleBindings when deployed
- The operator was only watching the namespace it was deployed
  into; that prevented it from reacting to the creation of
  AkkaCluster instances in other namespaces, in accordance with
  the OperatorGroup. That is specified in the 'env' section,
  in which the following appeared:
  ```
    env:
    - name: WATCH_NAMESPACE
      valueFrom:
        fieldRef:
          fieldPath: metadata.namespace
  ```
  That needs to be instead:
  ```
    env:
    - name: WATCH_NAMESPACE
      valueFrom:
        fieldRef:
          fieldPath: metadata.annotations['olm.targetNamespaces']
  ```

marcoderama commented:
Closing.  Merging to a new `lightbend` branch now in https://github.com/lightbend/community-operators/pull/4.

Commit:
Fixes to the ClusterServiceVersion; now working as intended.

A number of changes are included in this pull request:
- The 'required' section of the owned crds was redundant; also,
  parsing it would cause a bizarre "couldn't parse plural.group"
  error message
- The description and displayName are required for all the elements
  of specDescriptors and statusDescriptors (they were in the wrong
  place). Also, tried to give informative description strings
- the 'clusterPermissions' section was set to null; it needs to
  be a (possibly empty) array, instead
- the serviceAccount permissions were set in the 'permissions'
  section, which translates to RBAC roleBindings. This is not
  sufficient, since the operator needs to be able to work on
  namespaces other than the one it is deployed to. So, moved
  those definitions to 'clusterPermissions', instead, which
  get translated to clusterRoleBindings when deployed
- The operator was only watching the namespace it was deployed
  into; that prevented it from reacting to the creation of
  AkkaCluster instances in other namespaces, in accordance with
  the OperatorGroup. That is specified in the 'env' section,
  in which the following appeared:
  ```
    env:
    - name: WATCH_NAMESPACE
      valueFrom:
        fieldRef:
          fieldPath: metadata.namespace
  ```
  That needs to be instead:
  ```
    env:
    - name: WATCH_NAMESPACE
      valueFrom:
        fieldRef:
          fieldPath: metadata.annotations['olm.targetNamespaces']
  ```
